diff -Naur ncurses-6.1/ncurses/curses.priv.h ncurses-6.1_emscripten/ncurses/curses.priv.h
--- ncurses-6.1/ncurses/curses.priv.h	2018-01-20 23:13:41.000000000 -0300
+++ ncurses-6.1_emscripten/ncurses/curses.priv.h	2019-08-08 21:50:05.174897464 -0300
@@ -64,6 +64,23 @@
 #include <stddef.h>		/* for offsetof */
 #include <stdlib.h>
 #include <string.h>
+
+#ifdef __EMSCRIPTEN__ 
+#ifdef getenv
+#undef getenv
+#endif
+
+#define getenv(name)(char*)\
+({\
+  char* data = getenv(name);\
+  int length = strlen(data);\
+  char* dest = (char*)malloc(length+1);\
+  strncpy(dest, data, length);\
+  dest[length+1] = '\0';\
+  dest;\
+})
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 
